R/other.R

```diff
@@ -18,17 +18,22 @@ fct_other <- function(f, keep, drop, other_level = "Other") {
   f <- check_factor(f)
   check_exclusive(keep, drop)
 
-  levels <- levels(f)
   if (!missing(keep)) {
-    levels[!levels %in% keep] <- other_level
+    lvls_other(f, levels(f) %in% keep, other_level)
   } else {
-    levels[levels %in% drop] <- other_level
+    lvls_other(f, !levels(f) %in% drop, other_level)
   }
+}
 
-  if (!other_level %in% levels) {
-    return(f)
+# Replace specified levels (if any), with other.
+# @param keep A logical vector the same length as `levels(f)`
+lvls_other <- function(f, keep, other_level = "Other") {
```

Comment:
I also added this new helper function to ensure that the implementation of "othering" is identical for all code paths.